lib/client/api.go

```diff
@@ -953,9 +956,12 @@ func (tc *TeleportClient) ConnectToProxy() (*ProxyClient, error) {
 		return makeProxyClient(sshClient, m), nil
 	}
 	// we have exhausted all auth existing auth methods and local login
-	// is disabled in configuration
-	if tc.Config.SkipLocalAuth {
-		return nil, trace.BadParameter("failed to authenticate with proxy %v", proxyAddr)
+	// is disabled in configuration, or the user refused connecting to untrusted hosts
+	if tc.Config.SkipLocalAuth || tc.localAgent.UserRefusedHosts() {
```

Comment:
It might be useful to have these in separate blocks then return different errors. Knowing if authentication failed because because all auth methods were exhausted or because of an invalid host certificate would be useful when debugging issues.